c10/core/TensorImpl.cpp

```diff
@@ -1212,6 +1212,27 @@ void TensorImpl::empty_tensor_restride_symint(MemoryFormat memory_format) {
   // recompute contiguous flag, as currently NHWC/NCHW flags are not mutually
   // exclusive see #24090
   refresh_contiguous();
+  // hard code some known true settings, for unbacked case
+  // TODO: avoid chundering into the guards for computing these
+  switch (memory_format) {
```

Comment:
if you don't set true, do they default false?

Reply:
no, they would be set to some complicated expression from the guard-less bool implementation that would evaluate to true (often, sympy cannot figure it out though)